pyccl/halos/profiles/profile_base.py

```diff
@@ -47,35 +47,25 @@ def __init__(self):
                             "_real or _fourier implementation.")
         self.precision_fftlog = FFTLogParams()
 
-    @property
-    @abstractmethod
-    def normprof(self) -> bool:
-        """Normalize the profile in auto- and cross-correlations by
-        :math:`I^0_1(k\\rightarrow 0, a|u)`
-        (see :meth:`~pyccl.halos.halo_model.HMCalculator.I_0_1`).
-        """
-
-    # TODO: CCLv3 - Rename & allocate _normprof_bool to the subclasses.
+    def normalization(self, hmc, cosmo, a):
+        """Returns the overall normalisation factor of this profile.
 
-    def _normprof_false(self, hmc, **settings):
-        """Option for ``normprof = False``."""
-        return lambda *args, cosmo, a, **kwargs: 1.
-
-    def _normprof_true(self, hmc, k_min=1e-5):
-        """Option for ``normprof = True``."""
-        # TODO: remove the first two lines in CCLv3.
-        k_hmc = hmc.precision["k_min"]
-        k_min = k_hmc if k_hmc != k_min else k_min
-        M, mass_def = hmc._mass, hmc.mass_def
-        return functools.partial(self.fourier, k=k_min, M=M, mass_def=mass_def)
+        Args:
+            hmc (:class:`~pyccl.halos.HMCalculator`): a halo model calculator
+                object.
+            cosmo (:class:`~pyccl.core.Cosmology`): a Cosmology object.
+            a (float): scale factor.
 
-    def _normalization(self, hmc, **settings):
-        """This is the API adapter and it decides which norm to use.
-        It returns a function of ``cosmo`` and ``a``. Optional args & kwargs.
+        Reurns:
+            float: normalisation of this profile.
         """
-        if self.normprof:
-            return self._normprof_true(hmc, **settings)
-        return self._normprof_false(hmc, **settings)
+        uk0 = self.fourier(cosmo=cosmo, k=hmc.precision['k_min'],
+                           M=hmc._mass, a=a, mass_def=hmc.mass_def)
+        hmc._get_ingredients(cosmo, a, get_bf=False)
+        return hmc._integrate_over_mf(uk0)
+        # TODO: CCLv3 replace by the below in v3 (profiles will all have a
+        # default normalisation of 1. Normalisation will always be applied).
+        # return lambda *args, cosmo, a, **kwargs: 1.
```

Comment:
What does that mean? Won't the default be the integration over mf any more?

Reply:
No, the default will be 1 (no normalisation)

Reply:
Hummm... maybe it would be better to default to nothing, and make this an abstract method so that devs are forced to implement a normalization? What do you think? Just bringing this to the table, but I'm happy with the way it is now.

Reply:
The very much most common case will be no normalization, so I'd rather make that the default.